flink-runtime/src/main/java/org/apache/flink/runtime/deployment/PartitionInfo.java

```diff
@@ -143,7 +147,14 @@ public static PartitionInfo fromEdge(ExecutionEdge edge, SimpleSlot consumerSlot
 			}
 		}
 
-		return new PartitionInfo(partitionId, producerExecutionId, producerLocation, producerAddress);
+		PartitionInfo partitionInfo = new PartitionInfo(partitionId, producerExecutionId,
+				producerLocation, producerAddress);
+
+		if (LOG.isDebugEnabled()) {
```

Comment:
Just small nit, since we are now using SLF4J, we no longer need to do guard for simple debug logging [1]. Hopefully to get the code less busy =)

[1] http://slf4j.org/faq.html#logging_performance